modules/fundamental/src/error.rs

```diff
@@ -51,32 +51,53 @@ impl ResponseError for Error {
             Self::Purl(err) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("InvalidPurlSyntax", err))
             }
-            Self::Actix(err) => {
-                HttpResponse::InternalServerError().json(ErrorInformation::new("System Actix", err))
-            }
             Self::BadRequest { msg, status } => {
                 HttpResponse::build(*status).json(ErrorInformation::new("Bad request", msg))
             }
-            Self::Any(err) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("System unknown", err)),
             Error::Ingestor(inner) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("Ingestor error", inner))
             }
-            Error::UnsupportedHashAlgorithm => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Unsupported hash algorithm", "")),
-            Error::Storage(inner) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Unsupported hash algorithm", inner)),
-            Error::Database(err) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Database error", err)),
             Error::Query(err) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("Query error", err))
             }
             Error::IdKey(err) => HttpResponse::BadRequest().json(ErrorInformation::new("Key", err)),
             Error::StorageKey(err) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("Storage Key", err))
             }
-            Error::Data(msg) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Data-model corruption", msg)),
+
+            // The following match arms handle internal system errors that are not expected to occur.
+            // They are logged and a generic error response is returned to avoid leaking
+            // internal state to end users.
+            Self::Actix(err) => {
+                log::error!("System Actix: {}", err);
+                HttpResponse::InternalServerError()
+                    .json(ErrorInformation::new("Internal Server Error", ""))
```

Comment:
The json isn't really adding anything, so I think you can just...
```suggestion
                HttpResponse::InternalServerError().finish()
```

Reply:
I do tend to like APIs that have consistent response bodies on all errors, including 500s.